### PR TITLE
Refactored out WindowBase from TopLevel

### PR DESCRIPTION
**Pull request by kekekeks**
diagram

### CODE REVIEW

File: src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs [hunk 1]
```diff
@@ -97,7 +97,11 @@ public override double ViewportValue
         /// <inheritdoc/>
         public override Size MeasureOverride(Size availableSize)
         {
-            var window = Owner.GetVisualRoot() as TopLevel;
+            var maxAvailableSize = (Owner.GetVisualRoot() as WindowBase)?.PlatformImpl?.MaxClientSize;
+            if (!maxAvailableSize.HasValue)
+            {
+                maxAvailableSize = (Owner.GetVisualRoot() as TopLevel)?.ClientSize;
```
Comment (grokys) [inline src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs]: I'm not sure this logic is correct: won't this prevent a window that has a `SizeToContent` of anything other than `Manual` from resizing to fit the visible items?

File: src/Avalonia.Controls/Platform/IWindowBaseImpl.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        /// </summary>
+        IPlatformHandle Handle { get; }
+        
+        /// <summary>
+        /// Gets the maximum size of a window on the system.
+        /// </summary>
+        Size MaxClientSize { get; }
+
+        /// <summary>
+        /// Gets the client size of the toplevel.
```
Comment (grokys) [inline src/Avalonia.Controls/Platform/IWindowBaseImpl.cs]: s/Gets/Sets

File: src/Avalonia.Controls/Platform/IWindowBaseImpl.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        /// </summary>
+        void Resize(Size size);
```
Comment (grokys) [inline src/Avalonia.Controls/Platform/IWindowBaseImpl.cs]: Needs a doc comment for the size parameter. Also maybe call this `clientSize` just to emphasize the fact it's the client size, not the size including borders.

**Review by grokys — CHANGES_REQUESTED** [review]

**Comment by grokys** [comment]
Other than the item virtualization change (which i added as a single comment instead of part of the review, sorry) - looks good!

File: src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs (hunk 1 above)
Comment (kekekeks) [inline src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs]: I've only added the logic that will be executed if there is *no* window available. In this case it tries to obtain information from another type of toplevel.

**Review by grokys — APPROVED** [review]

**Pull request merged**